src/main/java/org/generationcp/middleware/service/api/permission/PermissionServiceImpl.java

```diff
@@ -56,19 +57,21 @@ private void getLinks(final Set<PermissionDto> permissionDtoList, final Permissi
 	}
 
 	@Override
-	public Permission getPermissionById(final Integer permissionId) {
-		return this.workbenchDaoFactory.getPermissionDAO().getById(permissionId);
+	public List<PermissionDto> getPermissionsDtoByIds(final Set<Integer> permissionIds) {
+		final List<PermissionDto> permissionDtoList = new ArrayList<>();
+		this.workbenchDaoFactory.getPermissionDAO().getPermissions(permissionIds).forEach( p -> {
+			final PermissionDto permissionDto = new PermissionDto(p);
+			final Map<RoleTypeDto, Boolean> roleTypeDtoSelectableMap = new HashMap<>();
+			p.getRoleTypePermissions().forEach(rtp -> {
+				roleTypeDtoSelectableMap.putIfAbsent(new RoleTypeDto(rtp.getRoleType()), rtp.getSelectable());
+			});
+			permissionDto.setRoleTypeSelectableMap(roleTypeDtoSelectableMap);
+			permissionDtoList.add(permissionDto);
+		}
+		);
+		return permissionDtoList;
```

Comment:
I feel like we can probably improve this, right now permissionDto.getRoleTypeSelectableMap is totally dependent on getPermissionsDtoByIds.
It is true that it is convenient because we are iterating once and building the map accordingly and for each permission, though from the method caller perspective this is not evident.

What about putting it as a separate service method instead of inside the dto?
```
Map<Integer, Map<RoleTypeDto, Boolean>> getRoleTypeSelectableByPermissionId()
```
or changing getPermissionsDtoByIds to make it more evident: `getPermissionsDtoByIdsWithRoleTypeSelectableMap`, a bit long but..
Or just a javadoc comment in each method stating the dependency..